mne/decoding/time_gen.py

```diff
@@ -914,18 +545,692 @@ def _predict(X, estimators):
+        scikit-learn Scorer instance.
+    scores_ : list of lists of float
+        The scores estimated by self.scorer_ at each training time and each
+        testing time (e.g. mean accuracy of self.predict(X)). Note that the
+        number of testing times per training time need not be regular;
+        else, np.shape(scores) = (n_train_time, n_test_time).
+
+
+    Notes
+    -----
+    The function implements the method used in:
+
+        Jean-Remi King, Alexandre Gramfort, Aaron Schurger, Lionel Naccache
+        and Stanislas Dehaene, "Two distinct dynamic modes subtend the
+        detection of unexpected sounds", PLoS ONE, 2014
+        DOI: 10.1371/journal.pone.0085791
+
+    .. versionadded:: 0.9.0
+    """
+    def __init__(self, picks=None, cv=5, clf=None, train_times=None,
+                 test_times=None, predict_mode='cross-validation', scorer=None,
+                 n_jobs=1):
+        super(GeneralizationAcrossTime, self).__init__(
+            picks=picks, cv=cv, clf=clf, train_times=train_times,
+            test_times=test_times, predict_mode=predict_mode, scorer=scorer,
+            n_jobs=n_jobs)
+
+    def __repr__(self):
+        """Human readable representation"""
+        s = ''
+        if hasattr(self, "estimators_"):
+            s += "fitted, start : %0.3f (s), stop : %0.3f (s)" % (
+                self.train_times_['start'], self.train_times_['stop'])
+        else:
+            s += 'no fit'
+        if hasattr(self, 'y_pred_'):
+            s += (", predicted %d epochs" % len(self.y_pred_[0][0]))
+        else:
+            s += ", no prediction"
+        if hasattr(self, "estimators_") and hasattr(self, 'scores_'):
+            s += ',\n '
+        else:
+            s += ', '
+        if hasattr(self, 'scores_'):
+            s += "scored"
+            if callable(self.scorer_):
+                s += " (%s)" % (self.scorer_.__name__)
+        else:
+            s += "no score"
+
+        return "<GAT | %s>" % s
+
+    def fit(self, epochs, y=None):
+        """ Train a classifier on each specified time slice.
+
+        Note. This function sets the ``picks_``, ``ch_names``, ``cv_``,
+        ``y_train``, ``train_times_`` and ``estimators_`` attributes.
+
+        Parameters
+        ----------
+        epochs : instance of Epochs
+            The epochs.
+        y : list or ndarray of int, shape (n_samples,) or None, optional
+            To-be-fitted model values. If None, y = epochs.events[:, 2].
+
+        Returns
+        -------
+        self : GeneralizationAcrossTime
+            Returns fitted GeneralizationAcrossTime object.
+
+        Notes
+        ------
+        If X and y are not C-ordered and contiguous arrays of np.float64 and
+        X is not a scipy.sparse.csr_matrix, X and/or y may be copied.
+
+        If X is a dense array, then the other methods will not support sparse
+        matrices as input.
+        """
+        return super(GeneralizationAcrossTime, self).fit(epochs, y=y)
```

Comment:
you don't need to add these fit and predict methods if they are implemented in the base class